File: DTOs/Sources/MeetingServiceDTOs/GetMeetingDTO.swift
import Foundation

public struct GetMeetingDTO: Codable {
    public var id: UUID
    public var name: String
    public var description: String
    public var status: MeetingStatus
    public var start: Date
    public var duration: UInt16?
    public var location: GetLocationDTO?
    public var chair: GetIdentityDTO?
    public var code: String?
    public var myAttendanceStatus: AttendanceStatus?
    
    public init(id: UUID, name: String, description: String, status: MeetingStatus, start: Date, duration: UInt16? = nil, location: GetLocationDTO? = nil, chair: GetIdentityDTO? = nil, code: String? = nil, myAttendanceStatus: AttendanceStatus? = nil) {
        self.id = id
        self.name = name
        self.description = description
        self.status = status
        self.start = start
        self.duration = duration
        self.location = location
        self.chair = chair
        self.code = code
        self.myAttendanceStatus = myAttendanceStatus
    }
}

public enum MeetingStatus: String, Codable, CaseIterable {
    case scheduled
    case inSession
    case completed
}

File: android/app/src/main/java/net/ipv64/kivop/dtos/MeetingServiceDTOs/GetMeetingDTO.kt
package net.ipv64.kivop.dtos.MeetingServiceDTOs

import java.time.LocalDateTime
import java.util.UUID

data class GetMeetingDTO(
    var id: UUID,
    var name: String,
    var description: String,
    var status: MeetingStatus,
    var start: LocalDateTime,
    var duration: UShort?,
    var location: GetLocationDTO?,
    var chair: GetIdentityDTO?,
    var code: String?,
    var myAttendanceStatus: AttendanceStatus?,
)

enum class MeetingStatus {
  scheduled,
  inSession,
  completed,
}
